### DISCUSSION TITLE
Migrate extensions to the new "expression and condition"

### DISCUSSION THREAD

**Discussion by D8H** [4ian/GDevelop#4533] [Enhancement ideas (and their technical discussions)]
# Operators
With the release 150, event-based extensions can use operators. It will be interesting to upgrade most of the extensions. I propose that we follow the same method to be consistent and keep the extensions compatible with existing events.

![image](https://user-images.githubusercontent.com/2611977/201074052-60df5c51-fa4a-4549-81df-39b9e22a41cd.png)

## Replace the expression by an "expression and condition"

The resulting expression with be the same and fully compatible. It only adds a condition for it.

![image](https://user-images.githubusercontent.com/2611977/201073499-767c39c3-25c1-4ede-98b5-e304a85a86bb.png)

## Add an "action with operator"

The action can't be replaced directly because parameters are not the same. We need to add a new one. I suggest that we use the "Op" suffix.

Tip: duplicate the existing action to avoid to copy-paste the events.

![image](https://user-images.githubusercontent.com/2611977/201073548-e5b6577f-5906-4c98-90b3-b4b68307d45e.png)

## Hide the existing action

The old action must be kept for compatibility reason. It must be hidden and can call the new action.

![image](https://user-images.githubusercontent.com/2611977/201073531-6bb04266-bde6-455d-84cb-bf84298ee529.png)

**Comment by tristanbob** [comment]
Does the "Action with operator" automatically adjust the value of the property?
Or do we need to add events to change the value?
If it is automatic, is there any way to clamp the values?

**Reply by D8H** [reply]
There is no generation for properties yet.
If you mean parameters, when the operator "+" is used, the "Value" parameter is the resulting value of the addition. The "related expression and condition is used automagically.
